### ISSUE TITLE
Very slow elements list for finitely presented semigroups isomorphic to finite semigroups

### ISSUE THREAD

**Issue by james-d-mitchell** [semigroups/Semigroups#425]
In the following example, the last line runs for a very long time, but it shouldn't, because the isomorphic partial permutation semigroup already knows its list of elements. 
~~~~
gap> S;
<inverse partial perm semigroup of size 103601, rank 8 with 8 generators>
gap> IsomorphismFpSemigroup(S);
MappingByFunction( <inverse partial perm semigroup of size 103601, rank 8
 with 8 generators>, <fp semigroup on the generators
[ s1, s2, s3, s4, s5, s6, s7, s8, s9, s10, s11, s12, s13, s14, s15, s16
 ]>, function( x ) ... end, function( x ) ... end )
gap> T := Range(last);
<fp semigroup on the generators [ s1, s2, s3, s4, s5, s6, s7, s8, s9, s10,
  s11, s12, s13, s14, s15, s16 ]>
gap> Elements(T); 
~~~~

**Comment by wilfwilson** [comment]
How would you resolve this?

**Comment by james-d-mitchell** [comment]
I’ve started to resolve this in Libsemigroups v1.0.0. Basically we need to improve the functionality for quotient semigroups in general, and resolving this issue will be part of that. @wilfwilson

**Comment by james-d-mitchell** [comment]
This specific issue is resolved in v4.0.0, via `NiceMonomorphism`s. It probably requires a bit more thorough of a solution, but I think I can close this issue now.
